src/fuselage/CCPACSSkinSegment.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+* Copyright (c) 2018 Airbus Defence and Space and RISC Software GmbH
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "CCPACSSkinSegment.h"
+
+#include <TopoDS_Face.hxx>
+#include <TopoDS_Edge.hxx>
+#include <BRep_Tool.hxx>
+#include <BRepExtrema_ExtCC.hxx>
+#include <TopExp.hxx>
+#include <BRepTools.hxx>
+#include <Bnd_Box.hxx>
+#include <BRepBndLib.hxx>
+#include <TopTools_IndexedMapOfShape.hxx>
+#include <Geom_Surface.hxx>
+
+#include "generated/CPACSSkinSegments.h"
+#include "CCPACSFrame.h"
+#include "CCPACSFuselageStringer.h"
+#include "CTiglError.h"
+#include "CCPACSFuselage.h"
+#include "CTiglUIDManager.h"
+#include "CNamedShape.h"
+
+namespace tigl
+{
+CCPACSSkinSegment::CCPACSSkinSegment(CCPACSSkinSegments* parent, CTiglUIDManager* uidMgr)
+    : generated::CPACSSkinSegment(parent, uidMgr)
+{
+}
+
+void CCPACSSkinSegment::Invalidate()
+{
+    m_borderCache = boost::none;
+}
+
+bool CCPACSSkinSegment::Contains(const TopoDS_Face& face)
+{
+    double u_min = 0., u_max = 0., v_min = 0., v_max = 0.;
+    BRepTools::UVBounds(face, u_min, u_max, v_min, v_max);
+    Handle(Geom_Surface) testSurf = BRep_Tool::Surface(face);
+    gp_Pnt point                  = testSurf->Value(u_min + ((u_max - u_min) / 2), v_min + ((v_max - v_min) / 2));
+    return Contains(point);
+}
+
+bool CCPACSSkinSegment::Contains(const TopoDS_Edge& edge)
+{
+    double u_min = 0., u_max = 0.;
+    Handle(Geom_Curve) curve = BRep_Tool::Curve(edge, u_min, u_max);
+    gp_Pnt point             = curve->Value(u_min + ((u_max - u_min) / 2));
+    return Contains(point);
+}
+
+bool CCPACSSkinSegment::Contains(const gp_Pnt& point)
+{
+    if (!m_borderCache)
+        UpdateBorders();
+
+    double x_mn = 0, x_mx = 0, y_mn = 0, y_mx = 0, z_mn = 0, z_mx = 0;
+    Bnd_Box boundingBox;
+    BRepBndLib::Add(m_parent->GetParent()->GetParent()->GetParent()->GetLoft()->Shape(), boundingBox);
+    boundingBox.Get(x_mn, y_mn, z_mn, x_mx, y_mx, z_mx);
+
+    double ym = (y_mx + y_mn) / 2;
+
+    const BorderCache& c = m_borderCache.value();
```

Comment:
Isn't this algorithm the same as https://github.com/DLR-SC/tigl/blob/3f311568a670529e934a4895163a933f2db6196a/src/wing/CCPACSWingCell.cpp#L138
?

If yes, we should only use one code. WingCell already contains unit tests, so it is known to be working.